File: poo/python/Interfaces/Conta/ContaPoupanca.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 12/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

from Conta import Conta

class ContaPoupanca(Conta):

    # Construtor #
    def __init__(self, numero):
        super(ContaPoupanca, self).__init__(numero)

    def atualiza(self, taxa):
        pai = super(ContaPoupanca, self)
        pai.setSaldo(pai.getSaldo() + 3 * pai.getSaldo() * taxa)

File: poo/java/Collecions/Exercicio5/ContaPoupanca.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 04/04/14
 *  R.A.: 86492
 */

import java.lang.Comparable;

public class ContaPoupanca extends Conta implements Comparable<ContaPoupanca>{

    /* Atributos */
    private String nomeDoCliente;

    /* Construtor */
    public ContaPoupanca(int numero, String nomeDoCliente){
        this.numero = numero;
        this.nomeDoCliente = nomeDoCliente;
    }

    /* Getters */
    public String getNomeDoCliente(){
        return this.nomeDoCliente;
    }

    /* Implementacao da interface */
    public int compareTo(ContaPoupanca aux){
        return this.getNomeDoCliente().compareTo(aux.getNomeDoCliente());
    }

    @Override
    public void atualiza(double taxa){
        this.saldo += 3*(this.saldo * taxa);
    }

    @Override
    public void imprime(){
        System.out.println("Conta " + this.numero + ": " +
                           "titular = " + this.getNomeDoCliente() +
                           " saldo = " + this.saldo );
    }
}
